src/amo/components/InstallButtonWrapper/index.js

```diff
@@ -50,128 +56,200 @@ type PropsFromState = {|
+          <a className="InstallButtonWrapper-download-link" href={installURL}>
+            {i18n.gettext('Download file')}
+          </a>
+        </div>
+      );
+    };
+
+    const overlayClassName = 'InstallButtonWrapper-QRCode-modal';
 
-  const showFileLink = () => {
     return (
-      <div className="InstallButtonWrapper-download">
-        <a className="InstallButtonWrapper-download-link" href={installURL}>
-          {i18n.gettext('Download file')}
-        </a>
-      </div>
+      addon && (
+        <div className="InstallButtonWrapper">
+          {!showLinkInsteadOfButton && (
+            <>
+              {showInstallButton ? (
+                <div>
+                  <AMInstallButton
+                    addon={addon}
+                    canUninstall={canUninstall}
+                    className={makeClassName(
+                      className ? `AMInstallButton--${className}` : '',
+                      {
+                        'AMInstallButton--noDownloadLink': !showDownloadLink,
+                      },
+                    )}
+                    currentVersion={currentVersion}
+                    defaultButtonText={defaultButtonText}
+                    disabled={!isCompatible}
+                    enable={enable}
+                    hasAddonManager={hasAddonManager}
+                    install={install}
+                    isAddonEnabled={isAddonEnabled}
+                    puffy={puffy}
+                    setCurrentStatus={setCurrentStatus}
+                    status={installStatus}
+                    uninstall={uninstall}
+                  />
+                  {_config.get('addonIdsWithQRCodes').includes(addon.id) &&
+                    clientApp !== CLIENT_APP_ANDROID &&
+                    _config.get('enableFeatureAddonQRCode') && (
+                      <div>
+                        <Button
+                          buttonType="none"
+                          className="InstallButtonWrapper-showQRCode-button"
+                          onClick={this.onShowQRCode}
+                          type="button"
+                        >
+                          {i18n.gettext(
+                            'Also available on Firefox for Android',
+                          )}
+                        </Button>
+                        {this.state.showQRCode && (
+                          <OverlayCard
+                            onEscapeOverlay={this.onHideQRCode}
+                            className={overlayClassName}
+                            id={overlayClassName}
+                            visibleOnLoad
+                          >
+                            <AddonQRCode
+                              addon={addon}
+                              onDismiss={this.onHideQRCode}
+                            />
+                          </OverlayCard>
+                        )}
+                      </div>
+                    )}
+                </div>
+              ) : null}
```

Comment:
This should be a component in itself. The `InstallButtonWrapper` is quite big already.